Fix warnings occuring during lint

Delgee opened the pull request:
I tried to remove all warnings happening in linting on cdm. Unfortunately `nx/eslint-plugin` latest version has only `@typescript-eslint:v5.60.1` as a peer dependency.

Edit: As @Pl217 mentioned below it was possible to install `@typescript-eslint:v5.61.0`. Updated `@nx` packages to remove all warnings happening in linting.

Delgee committed:
Update `assignment` to `originalAssignment`

On `useEffect` hook `assignment` variable was always using `originalAssignment` since `updatedAssignment`
was not initialized yet. To clean up dependency array used `originalAssignment` instead of `assignment`.

unocha-hpc reviewed (approved):
Checks have passed and this pull request is ready for manual review

Pl217 reviewed (changes requested):
Good work, but you're lacking some detail:
- In the first comment (PR comment, not commit) you said: "I tried to remove all warnings happening in linting on cdm. Unfortunately `nx/eslint-plugin` latest version has only `@typescript-eslint:v5.60.1` as a peer dependency". You are missing information about what this fact prevents you from doing. Looks like you want to continue that sentence with a comma and then "which prevents us from installing `v5.61.0`, that adds official support for Typescript 5.1". I checked and `@nx/eslint-plugin` [specifies `^5.60.1` (with caret)](https://github.com/nrwl/nx/blob/d42d2f46fbad6171288d04c864e54cfae4318969/packages/eslint-plugin/package.json#L27) as a peer dependency. The fact that they are using a caret should allow you to install `v5.61.0`.
- This PR is in CDM repo, so, you don't need to reference CDM in the PR title
- Commit messages usually have their title and body separated by a blank line. Some tools forgive you this, like GitHub, you see how they collapse everything after the first line. However, if you don't have an empty line between title and body, everything will be considered as title. Use `git log --decorate --oneline --graph` in you terminal to see this in action.
- You didn't explain anywhere why you're changing `useState` for `useRef` when defining XForm. This is mixed into first commit and you should probably commit this change separately, explaining the reason behind change in commit message

Edit: Looks like you can avoid switching to `useRef`, if you rewrite `useEffect` into two:
```ts
// Clean up XForm when it is unmounted
useEffect(() => {
  return () => {
    if (xform) {
      xform.resetView();
    }
  };
}, [xform]);

// Clear Enketo cache (or something going on there which slows down
// form re-init/getData) when this component is unmounted
useEffect(() => {
  return () => {
    navigate(0);
  };
}, [navigate]);
```

Delgee committed:
Remove deprecated `nx workspace-lint` command

unocha-hpc reviewed (approved):
Checks have passed and this pull request is ready for manual review

Delgee committed:
Don't use `useState` hook in `xform`

Since `xform` variables value is not used in render it's better to use
`useRef` hook instead of `useState` hook

Delgee committed:
Fix missing dependencies in `useEffect` hook

By declaring `unMount` function inside `useEffect` it will be no longer
necessary to add `unMount` to the dependency array.

unocha-hpc reviewed (approved):
Checks have passed and this pull request is ready for manual review

Delgee commented:
In my opinion, using the `useRef` hook would be a better choice compared to `useState`. According to the official React documentation, the `useRef` hook is [React Hook that lets you reference a value that’s not needed for rendering.](https://react.dev/reference/react/useRef) In this case, since the value of the `xform` variable is not needed for rendering and should not change, using it as a stateful variable would be incorrect.

Pl217 commented on package.json:
```diff
@@ -65,13 +65,13 @@
+    "@types/react": "18.2.13",
+    "@types/react-dom": "18.2.6",
     "@types/react-helmet": "6.1.6",
-    "@types/react-is": "18.2.0",
+    "@types/react-is": "18.2.1",
     "@types/styled-components": "5.1.26",
-    "@typescript-eslint/eslint-plugin": "5.59.8",
-    "@typescript-eslint/parser": "5.59.8",
+    "@typescript-eslint/eslint-plugin": "5.61.0",
+    "@typescript-eslint/parser": "5.61.0",
```
Because `@nx/eslint-plugin:v16.5.0` [has a peer dependency](https://github.com/nrwl/nx/blob/16.5.0/packages/eslint-plugin/package.json#L27) `"@typescript-eslint/parser": "^5.60.1"` npm install fails unless you update it to `>= 5.60.1`, which you are doing with update to `v5.61.0`, which [brings](https://github.com/typescript-eslint/typescript-eslint/releases/tag/v5.61.0) us official Typescript 5.1 support.

I understand you should update `@typescript-eslint/eslint-plugin` to match the same version, but why did you update a bunch of `@types/*` packages for React?

Delgee replied:
Since we are using `nx` I used `nx migrate` command to update `nx` plugins. So these updates are from automated migration.

Pl217 replied:
Ok, I only thought of migrations when doing major version upgrade of Nx, but you're right, using `nx migrate` is the correct way to upgrade. We've done plenty major version upgrades of Nx in the past (see #365 for latest one), and I don't like how `nx migrate` treats other dependencies. Sometimes, it can also downgrade our dependencies. In this case, I would keep the upgrade to React dependencies (I see that we can upgrade `@types/react` even more), but commit it separately from Nx upgrade.

Whether you manually change output of `nx migrate` (which I prefer) or not, could be a matter of "taste", but the actual problem is that you forgot to **run** migrations. `npx nx migrate latest` also creates `migrations.json` file (which should not be committed). After upgrading packages (in multiple commits like said), you should have run `npx nx migrate --run-migrations`. There is a minor change in `nx.json` caused by this migration.

Regarding commit messages, look at f62f3791af6a7ee359f7f1b8491098327fbeaf1e and efedfe9c787cb8a521b37e9b8979f9c009eba590 to see how I explained that commit was actually generated by Nx migration and in the second commit I even included console output from running a migration. In case of this update to 16.5.0, I've got this on my machine:

```
 >  NX   Running migrations from 'migrations.json'

Ran add-test-setup-to-inputs-ignore from @nx/jest
  Add test-setup.ts to ignored files in production input

  UPDATE nx.json
---------------------------------------------------------
```

unocha-hpc reviewed (approved):
Checks have passed and this pull request is ready for manual review

Pl217 reviewed (changes requested):
Last commit says:
> This commit is the output of running command `npx nx migrate --run-migrations`
> Commit contains full output of this command, without any manual changes.

Initially I thought "Cool, you can skip running `npx nx migrate latest` and directly execute the `migrate` with `--run-migrations`, where `latest` is implicit", but this isn't possible, as I get an error in console:
> File 'migrations.json' doesn't exist, can't run migrations. Use flag --if-exists to run migrations only if the file exists

So, you have to run `npx nx migrate latest` first, which generates a `migrate.json` file and then you run `npx nx migrate --run-migrations`.

Delgee committed:
⬆️ Update `nx` plugins to `v16.5.0`

This commit is generated by running the following command:
`npx nx migrate latest`

This has changed `package.json`

Besides package updates, `nx migrate` command also
generated `migrations.json` file, which should not
be committed, but will be used as a blueprint for
migrations in the next step, when we run
`npx nx migrate --run-migrations`

Delgee committed:
Migrate project to NX `v16.5.0`

This commit is the output of running command
`npx nx migrate --run-migrations`
Commit contains full output of this command,
without any manual changes.

Full output of command run:

>  NX   Running migrations from 'migrations.json'

Ran add-test-setup-to-inputs-ignore from @nx/jest
  Add test-setup.ts to ignored files in production input

  UPDATE nx.json

unocha-hpc reviewed (approved):
Checks have passed and this pull request is ready for manual review
